snarkos/ledger/mod.rs

```diff
@@ -125,6 +115,22 @@ impl<N: Network> Ledger<N> {
         // Return the next block.
         Ok(next_block)
     }
+
+    /// Attempts to add the given block to the ledger.
+    pub(crate) async fn add_next_block(self: &Arc<Self>, next_block: &Block<N>) -> Result<()> {
+        // Ensure the given block is a valid next block.
+        self.ledger.read().check_next_block(next_block)?;
+
```

Comment:
```suggestion
```
this operation is redundant, as it's already performed within `Ledger::add_next_block`: https://github.com/AleoHQ/snarkVM/blob/testnet3/vm/compiler/src/ledger/mod.rs#L539

Reply:
it's easy to miss such things, which is another (besides added safety) argument for introducing something like I mentioned here: https://github.com/AleoHQ/snarkVM/pull/990#issuecomment-1220581592

Reply:
if we want the block checks to be more explicit (i.e. outside of `add_next_block`) and to introduce some extra safety, the simplest solution would be something like [this](https://github.com/AleoHQ/snarkVM/compare/testnet3...ljedrz:snarkVM:explicit_block_verification?expand=1). This assumes that the `Block` is the point where all the verification happens (which makes sense).

Reply:
Great catch! For now, I've removed the duplicate check. We can update this again if we go down the route of passing through explicitly "Verified<Block>"s  in snarkVM.